pkg/TIPA.py

```
@@ -164,587 +165,498 @@ def __init__(self, img, mouse_pos, display_info_init, gui_queue):
+        try:
+            if site not in ["market", "wiki"]:
+                raise ValueError("Invalid site. Choose 'market' or 'wiki'.")
+
+            search_url = f"https://www.google.com/search?q=tarkov+{site}+" + urllib.parse.quote_plus(search_text)
+            page = requests.get(search_url)
+            page.raise_for_status()  # Raises an HTTPError if the status is 4xx, 5xx
+
+            soup = BeautifulSoup(page.content, 'html.parser')
+            search_results = soup.find("div", {"class": "egMi0 kCrYT"})
+            if not search_results:
+                raise Exception("No search results found.")
```

Ruff (0.8.0)

Abstract `raise` to an inner function

(TRY301)